pkg/sql/authorization.go

```diff
@@ -252,21 +261,48 @@ func (p *planner) CheckPrivilegeForUser(
 	// context of support escalations, we need to be able to grant the ability to
 	// view system tables without granting the entire admin role.
 	if d, ok := privilegeObject.(catalog.Descriptor); ok {
-		if catalog.IsSystemDescriptor(d) && privilegeKind == privilege.SELECT {
-			hasViewSystemTablePriv, err := p.HasPrivilege(
-				ctx, syntheticprivilege.GlobalPrivilegeObject, privilege.VIEWSYSTEMTABLE, user,
-			)
-			if err != nil {
-				return err
-			}
-			if hasViewSystemTablePriv {
-				return nil
+		if catalog.IsSystemDescriptor(d) {
```

Comment:
Here we apply the check for the system descriptors (eg, system.eventlog, system.raftlog, etc).